server/index-node/src/resolver.rs

```diff
@@ -191,6 +234,54 @@ where
         Ok(poi)
     }
 
+    fn resolve_public_proofs_of_indexing(
+        &self,
+        field: &a::Field,
+    ) -> Result<r::Value, QueryExecutionError> {
+        let requests = field
+            .get_required::<Vec<PublicProofOfIndexingRequest>>("requests")
+            .expect("valid requests required, validation should have caught this");
+
+        Ok(r::Value::List(
+            requests
+                .clone()
+                .iter()
+                .map(|request| {
+                    match futures::executor::block_on(
+                        self.store.get_public_proof_of_indexing(
```

Comment:
This is fine as long as we're not worried about the number of queries. `get_public_proof_of_indexing` runs 2 SQL queries, and we're calling it `requests.len()` times. Just checking if we're okay with this.

Reply:
You're right, this is a potential exploit, good catch. I'm going to limit the number of requests to 10 for now. Any sensible indexer would rate limit their status API anyway, and this is done by default in indexer-service.